types/errors/errors.go

```diff
@@ -394,3 +397,36 @@ type causer interface {
 type unpacker interface {
 	Unpack() []error
 }
+
+func GRPCWrap(err error, c codes.Code, msg string) error {
+	if err != nil {
+		st := status.New(c, msg)
+		var sdkErr *Error
+		if errors.As(err, &sdkErr) {
+			errorInfo := &errdetails.ErrorInfo{
+				Reason:   sdkErr.Error(),
+				Metadata: map[string]string{"Codespace": sdkErr.Codespace(), "ABCICode": fmt.Sprintf("%d", sdkErr.ABCICode())},
+			}
+			st, err = st.WithDetails(errorInfo)
+			if err != nil {
+				return err
+			}
+		}
+
+		stack := stackTrace(err)
+		var stackEntries []string
+		stackEntries = make([]string, len(stack))
+		for i, f := range stack {
+			stackEntries[i] = fmt.Sprintf("%+v", f)
+		}
+		dbgInfo := &errdetails.DebugInfo{
+			StackEntries: stackEntries,
+		}
+		st, err = st.WithDetails(dbgInfo)
+		if err != nil {
+			return err
+		}
+		return st.Err()
+	}
+	return nil
```

Comment:
this code gets a bit confusing at the end of the function, an early exit in at the beginning of the function would help
```go
func GRPCWrap(err error, c codes.Code, msg string) error {
    if err == nil {
        return nil
    }
    // enrich code below